src/SFA.DAS.ApplyService.Web/Controllers/Roatp/RoatpManagementHierarchyController.cs

```diff
@@ -0,0 +1,333 @@
+using SFA.DAS.ApplyService.Domain.Apply;
+using SFA.DAS.ApplyService.Session;
+using SFA.DAS.ApplyService.Web.Infrastructure;
+using SFA.DAS.ApplyService.Web.Services;
+using SFA.DAS.ApplyService.Web.Validators;
+using SFA.DAS.ApplyService.Web.ViewModels.Roatp;
+using SFA.DAS.ApplyService.Web.ViewModels.Roatp.ManagementHierarchy;
+
+namespace SFA.DAS.ApplyService.Web.Controllers.Roatp
+{
+    public class RoatpManagementHierarchyController  : RoatpApplyControllerBase
+    {
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly IApplicationApiClient _applicationApiClient;
+        private readonly IAnswerFormService _answerFormService;
+        private readonly ITabularDataRepository _tabularDataRepository;
+
+        public RoatpManagementHierarchyController(ISessionService sessionService, IQnaApiClient qnaApiClient, IApplicationApiClient applicationApiClient, IAnswerFormService answerFormService, ITabularDataRepository tabularDataRepository) : base(sessionService)
+        {
+            _qnaApiClient = qnaApiClient;
+            _applicationApiClient = applicationApiClient;
+            _answerFormService = answerFormService;
+            _tabularDataRepository = tabularDataRepository;
+        }
+
+        [Route("management-hierarchy")]
+        public async Task<IActionResult> StartPage(Guid applicationId)
+        {
+            var peopleData = await _qnaApiClient.GetAnswerByTag(applicationId, RoatpWorkflowQuestionTags.AddManagementHierarchy);
+
+            if (peopleData != null && peopleData.Value != null)
+            {
+                return await ConfirmManagementHierarchy(applicationId);
+            }
+
+            return AddManagementHierarchy(applicationId);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> ConfirmManagementHierarchy(Guid applicationId)
+        {
+            var ManagementHierarchyData = await _tabularDataRepository.GetTabularDataAnswer(applicationId, RoatpWorkflowQuestionTags.AddManagementHierarchy);
+
+            if (ManagementHierarchyData == null || ManagementHierarchyData.DataRows == null || ManagementHierarchyData.DataRows.Count == 0)
+            {
+                return RedirectToAction("AddManagementHierarchy", new { applicationId });
+            }
+
+            var model = new ConfirmManagementHierarchyViewModel { ApplicationId = applicationId, ManagementHierarchyData = ManagementHierarchyData };
+            PopulateGetHelpWithQuestion(model, "ConfirmManagementHierarchy");
+
+            return View("~/Views/Roatp/ManagementHierarchy/ConfirmManagementHierarchy.cshtml", model);
+        }
+
+
+
+        [HttpGet]
+        public IActionResult AddManagementHierarchy(Guid applicationId)
+        {
+            var model = new AddEditManagementHierarchyViewModel { ApplicationId = applicationId, GetHelpAction = "AddManagementHierarchy" };
+            
+            //MFCMFC not sure about the variable being injected here.....
```

Comment:
The page ID is used the store the get help question text in session against the page